### PR TITLE
Adds todo guidelines

### PR DESCRIPTION
**Pull request by JosephSalisbury**
Towards https://github.com/giantswarm/giantswarm/issues/4463

### CODE REVIEW

File: go/todo.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+and link it in the code.
+
+This helps ensure that TODOs don't die in the codebase,
+but are actioned by SIGs.
+
+e.g:
+```
+// TODO: Remove the foobar once the jank is lollobazzed.
+// See https://github.com/giantswarm/giantswarm/issues/9001
+```
```
Comment (xh3b4sd) [inline go/todo.md]: Here comes the opinion train. I did this so far. 

```
// TODO foo bar. 
// 
//     https://github.com/issue/link
// 
```

This looks nicer in godoc and the link is easier to catch with the eyes. The 4 character indentation is displayed in a special way in the browser as well.

**Review by xh3b4sd — APPROVED** [review]
Go with what you think is right.

File: go/todo.md (hunk 1 above)
Comment (kopiczko) [reply go/todo.md]: +1 To be compatible with godoc (even if it isn't a strict godoc) it should should be actually a tab. See https://golang.org/src/text/template/doc.go for example.

File: go/todo.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# TODOs
+
+When writing a TODO in code, create a TODO issue
+(see https://github.com/giantswarm/giantswarm/issues/new?template=todo.md)
+and link it in the code.
+
+This helps ensure that TODOs don't die in the codebase,
+but are actioned by SIGs.
```
Comment (r7vme) [inline go/todo.md]: -> it will die in github :D

**Review by r7vme — APPROVED** [review]

**Review by oponder — APPROVED** [review]

File: go/todo.md (hunk 1 above)
Comment (JosephSalisbury) [reply go/todo.md]: Done, with a tab. Bam!

**Pull request merged**